Dockerfile

```diff
@@ -15,44 +15,40 @@
 # Build the manager binary
 FROM golang:1.20 as builder
 
-ARG ARCH ?= $(shell go env GOARCH)
-ARG TARGETPLATFORM
-ENV TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
-ARG LDFLAGS
-ARG GO_BUILD_GC_ARGS
-
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod \
```

Comment:
This change requires buildx and is currently incompatible with using #757 as a docker engine on linux. I'm not sure where else buildx might be missing. 

`buildx` is the default and classic build is disabled.